engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/SchemaRegistryKafkaAvroProvider.scala

```diff
@@ -1,23 +1,24 @@
 package pl.touk.nussknacker.engine.avro.schemaregistry
 
+import cats.data.Validated
 import javax.annotation.Nullable
 import org.apache.flink.streaming.connectors.kafka.{KafkaDeserializationSchema, KafkaSerializationSchema}
 import pl.touk.nussknacker.engine.api.typed.typing
 import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
-import pl.touk.nussknacker.engine.avro.{AvroUtils, KafkaAvroSchemaProvider}
+import pl.touk.nussknacker.engine.avro.{AvroUtils, KafkaAvroException, KafkaAvroSchemaProvider}
 import pl.touk.nussknacker.engine.kafka.{KafkaConfig, RecordFormatter}
 
 class SchemaRegistryKafkaAvroProvider[T](schemaRegistryProvider: SchemaRegistryProvider[T],
                                          kafkaConfig: KafkaConfig,
                                          topic: String,
                                          version: Option[Int]) extends KafkaAvroSchemaProvider[T] {
 
-  override def typeDefinition: typing.TypingResult =
-    AvroSchemaTypeDefinitionExtractor.typeDefinition(
-      schemaRegistryProvider.createSchemaRegistryClient.getSchema(
-        AvroUtils.valueSubject(topic), version
-      )
-    )
+  override def typeDefinition: Validated[KafkaAvroException, typing.TypingResult] =
+    schemaRegistryProvider
+      .createSchemaRegistryClient
+      .getSchema(AvroUtils.valueSubject(topic), version)
+      .leftMap(clientError => KafkaAvroException(clientError.message))
```

Comment:
Do we need both KafkaAvroException and SchemaRegistryClientError?

Reply:
I will leave only one.